### PR TITLE
add selfsigned cert as k8s secret

### PR DESCRIPTION
**Pull request by kinarashah**
https://github.com/rancher/rancher/issues/13619

### CODE REVIEW

File: app/listenconfig_data.go [hunk 1]
```diff
@@ -75,11 +76,36 @@ func addListenConfig(management *config.ManagementContext, cfg Config) error {
...
+		}
+		secret.Name = "tls-rancher"
+		secret.Namespace = "cattle-system"
+
+		if _, err := management.Core.Secrets("cattle-system").Get("tls-rancher", v1.GetOptions{}); apierrors.IsNotFound(err) {
+			_, err = management.Core.Secrets("cattle-system").Create(secret)
+			return err
+		}
+
+		_, err = management.Core.Secrets("cattle-system").Update(secret)
```
Comment (alena1108) [inline app/listenconfig_data.go]: is there a way to compare the content to eliminate updates when nothing had changed?

File: app/listenconfig_data.go [hunk 2]
```diff
@@ -75,11 +76,36 @@ func addListenConfig(management *config.ManagementContext, cfg Config) error {
...
+		data := map[string]string{}
+		data["tls.key"] = cfg.ListenConfig.CAKey
+		data["tls.crt"] = cfg.ListenConfig.CACert
+
+		secret := &corev1.Secret{
+			StringData: data,
+			Type:       corev1.SecretTypeTLS,
+		}
+		secret.Name = "tls-rancher"
+		secret.Namespace = "cattle-system"
```
Comment (alena1108) [inline app/listenconfig_data.go]: nit: const for cattle namespace

File: app/listenconfig_data.go [hunk 3]
```diff
@@ -75,11 +76,36 @@ func addListenConfig(management *config.ManagementContext, cfg Config) error {
...
+			return err
+		}
+	} else {
+		cfg.ListenConfig.ResourceVersion = existing.ResourceVersion
+		if _, err := management.Management.ListenConfigs("").Update(cfg.ListenConfig); err != nil {
+			return err
+		}
+	}
+
+	if selfSigned {
```
Comment (alena1108) [inline app/listenconfig_data.go]: nit: can we return immediately if !selfSigned

File: app/listenconfig_data.go (hunk 1 above)
Comment (kinarashah) [reply app/listenconfig_data.go]: checked that `selfSigned` is set true only when new cert is generated, thus also covering the case of no updates

**Pull request merged**